src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseAuthenticationController.java

```diff
@@ -163,7 +163,8 @@ public BaseAuthenticationController(AuthenticationManager authenticationManager,
     private void checkIfUserBlocked(String username) throws PermissionDeniedException {
 
         if (loginAttemptService.isBlocked(username)) {
-            throw new PermissionDeniedException("You have exceeded the number of allowed login attempts. Please try again later.");
+            final String errorMessage = String.format("You have exceeded the number of allowed login attempts. Please try again in %s minutes.", loginAttemptService.getAttemptsResetMinutes());
```

Comment:
`attemptsResetMinutes` is maximum time to lock out account, but what if someone tried to login, got an error and come back after in two minutes and see the same message?

Reply:
Yes in this case he will see still the default number of minutes but also worth to take into account that guava cache does not provide functionality to determine exact remaining lifespan of the particular key entry so I cannot evaluate it without significant changes of the account lock functionality. 
First time users sees this message - it is exactly correct but farther run-time reevaluations are not supported.

Reply:
Why not to store timestamp in cache and calculate duration on the fly?

Reply:
@wivern, I have looked into code bit dipper and discovered that for the any unsuccessful login attempt we bump number of unsuccessful attempts which is update cache configured like:
cacheBuilder.newBuilder().
                expireAfterWrite(attemptsResetMinutes, TimeUnit.MINUTES).

See:
BaseAuthenticationController::getJsonResultForUnsuccessfulLogin

so any write to the cache refresh expiration time back to %arachne.loginAttempts.resetMinutes%

Taking this into account now I think current message is correct in all the cases.
That do you think?

Reply:
I have update the code not to refresh failedAttemps counter after reaching MAX number as it was not correct to bump counter  without making real login attempt.